src/function/scalar/map/map_extract.cpp

```diff
@@ -41,26 +79,30 @@ static void MapExtractFunction(DataChunk &args, ExpressionState &state, Vector &
 
 	UnifiedVectorFormat map_data;
 
-	auto &map_keys = MapVector::GetKeys(map);
-	auto &map_values = MapVector::GetValues(map);
+	// Create the chunk we'll feed to ListPosition
+	DataChunk list_position_chunk;
+	vector<LogicalType> chunk_types;
+	chunk_types.reserve(2);
+	chunk_types.push_back(map.GetType());
+	chunk_types.push_back(key.GetType());
+	list_position_chunk.InitializeEmpty(chunk_types.begin(), chunk_types.end());
 
-	map.ToUnifiedFormat(args.size(), map_data);
+	// Populate it with the map keys list and the key vector
+	list_position_chunk.data[0].Reference(map);
+	list_position_chunk.data[1].Reference(key);
+	list_position_chunk.SetCardinality(tuple_count);
 
-	for (idx_t row = 0; row < args.size(); row++) {
-		idx_t row_index = map_data.sel->get_index(row);
-		auto key_value = key.GetValue(row);
+	Vector position_vector(LogicalType::LIST(LogicalType::INTEGER), tuple_count);
+	// We can pass around state as it's not used by ListPositionFunction anyways
+	ListContainsOrPosition<int32_t, PositionFunctor, MapKeyArgFunctor>(list_position_chunk, position_vector);
 
-		list_entry_t entry = ListVector::GetData(map)[row_index];
-		auto offsets = MapVector::Search(map_keys, args.size(), key_value, entry);
```

Comment:
Is `MapVector::Search` unused after this PR?

Reply:
Ah yea now that you mention it, probably is
I'll clean up the dead code :+1: